### PR TITLE
Added AVL Tree Implementation

### PR DESCRIPTION
**Pull request by arnoob16**
## Have you read the Contribution Guidelines?

Yes

## Description

AVL Tree Implementation added in Java.

## Checklist
- [ x ] I've read the contribution guidelines.
- [ x ] I've checked the issue list before deciding what to submit.
- [ x ] I've edited the README.md and link to my code.
 
 
## Related Issues (Mention Issue no. like #xxx)

Fixes Issue 
<br/>
#327 
(Write your answer above.)

### CODE REVIEW

File: Java/AVL_Tree_Implementation.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        tree.preOrder(tree.root);
+        System.out.println("Enter the node to be deleted : ");
+        int delKey = sc.nextInt();
+		tree.root = tree.deleteNode(tree.root, delKey);
+		System.out.println("");
+		System.out.println("Preorder traversal after deletion of 10 :");
+        tree.preOrder(tree.root);
+        sc.close();
+	}
+}
```
Comment (mephi007) [inline Java/AVL_Tree_Implementation.java:176]: Add sample input, ouput, Time and Space Complexity

**Review by manan025 — CHANGES_REQUESTED** [review]
Please add sample I/O and resolve conflict.

**Review by arnoob16 — COMMENTED** [review]
Added Sample I/O.
Removed Conflict.

**Review by manan025 — CHANGES_REQUESTED** [review]
Please add time and space complexity in the comment as well. Everything else seems great!

**Comment by sukritishah15** [comment]
@divyanshkhatana - Please make the required changes TODAY.
As we will re-review this PR again TODAY.

Thank you!

**Comment by divyanshkhatana** [comment]
Can you please tell what all changes?

--
Sent from myMail for Android Monday, 05 October 2020, 01:55pm +05:30 from Sukriti Shah  notifications@github.com :

**Comment by arnoob16** [comment]
Hey @manan025 I've added the Complexity of my implementation as well.

**Review by sukritishah15 — APPROVED** [review]

**Comment by sukritishah15** [comment]
Nice work @arnoob16 !

**Comment by divyanshkhatana** [comment]
i have added the i/o and complexities

**Pull request merged**